packages/extensions/dataflow/src/lnode/lnode-element.vue

```diff
@@ -57,10 +63,28 @@ function onSelect(lnodeId: string) {
 	emit('change', lnodeId)
 }
 
-// Get the label for the port (e.g., "DataObject.Name.DataAttribute.Name")
-function getPortLabel(dataObject: any): string {
-	const daNames = (dataObject.dataAttributes ?? []).map((da: any) => da.name)
-	return [dataObject.name, ...daNames].join('.')
+function getPortCount(lnode: LNode): number {
+	// only DOS.DAS with subscriberLNode
+	return (
+		lnode.dataObjectSpecifications?.reduce(
+			(count, dos) =>
+				count +
+				(dos.dataAttributeSpecification?.filter((das) => !!das.subscriberLNode).length ?? 0),
+			0,
+		) ?? 0
+	)
+}
+
+// Port-Index-Function
+function portIdx(dosIdx: number, dasIdx: number) {
+	let idx = 0
+	if (!activeLNode.value?.dataObjectSpecifications) return 0
+	for (let i = 0; i < dosIdx; i++) {
+		idx += activeLNode.value.dataObjectSpecifications[i].dataAttributeSpecification.filter(
```

Comment:
I would prefer plural for the attribute name, because it is an array.
```suggestion
		idx += activeLNode.value.dataObjectSpecifications[i].dataAttributeSpecifications.filter(
```